force-app/main/default/classes/AffiliationMappingsService.cls

```diff
@@ -101,6 +101,47 @@ public virtual with sharing class AffiliationMappingsService {
         this.locateAffiliationMappingsMapper().updateAffiliationMappings(afflMappingModel);
     }
 
+    /*****************************************************************************
+     * @description Checks user permissions and update the auto enrollment mapppings.
+     *****************************************************************************/
+    public virtual void updateAutoEnrollmentMappings(
+        String mappingName,
+        String oldAccountRecordType,
+        String newAccountRecordType,
+        String autoProgramEnrollmentStatus,
+        String autoProgramEnrollmentRole
+    ) {
+        this.locateUserPermissionService().checkViewSetupForCurrentUser();
+        if (oldAccountRecordType != newAccountRecordType) {
+            List<AffiliationMappingsModel> afflMappingModels = this.locateAffiliationMappingsMapper()
+                .getAffiliationMappingsModel();
+            List<Affl_Mappings__c> affliliaitonMappingsToUpdateList = new List<Affl_Mappings__c>();
+
+            Affl_Mappings__c oldAfflSetting = Affl_Mappings__c.getInstance(mappingName);
+            oldAfflSetting.Auto_Program_Enrollment_Status__c = null;
+            oldAfflSetting.Auto_Program_Enrollment_Role__c = null;
+            oldAfflSetting.Auto_Program_Enrollment__c = false;
+            affliliaitonMappingsToUpdateList.add(oldAfflSetting);
+
+            for (AffiliationMappingsModel afflMappingModel : afflMappingModels) {
+                if (afflMappingModel.accountRecordTypeFieldValue == newAccountRecordType) {
+                    Affl_Mappings__c newAfflSetting = Affl_Mappings__c.getInstance(afflMappingModel.name);
+                    newAfflSetting.Auto_Program_Enrollment_Status__c = autoProgramEnrollmentStatus;
+                    newAfflSetting.Auto_Program_Enrollment_Role__c = autoProgramEnrollmentRole;
+                    newAfflSetting.Auto_Program_Enrollment__c = true;
+                    affliliaitonMappingsToUpdateList.add(newAfflSetting);
+                    break;
+                }
+            }
+            update affliliaitonMappingsToUpdateList;
+        } else {
+            Affl_Mappings__c afflSetting = Affl_Mappings__c.getInstance(mappingName);
+            afflSetting.Auto_Program_Enrollment_Status__c = autoProgramEnrollmentStatus;
+            afflSetting.Auto_Program_Enrollment_Role__c = autoProgramEnrollmentRole;
+            update afflSetting;
```

Comment:
This behavior should be moved to the mapper. We shouldn't be performing any DML operations in a service.